lib/eth/contract.rb

```diff
@@ -37,27 +37,58 @@ def initialize(name, bin, abi)
       @constructor_inputs, @functions, @events = parse_abi(abi)
     end
 
-    # Creates a contract wrapper.
+    # Creates a contract wrapper from a Solidity file.
     #
     # @param file [String] solidity file path.
-    # @param bin [String] contract bin string.
+    # @param contract_index [Number] specify contract.
+    # @return [Eth::Contract::Object] Returns the class of the smart contract.
+    # @raise [ArgumentError] if the file path is empty or no contracts were compiled.
+    def self.from_file(file: nil, contract_index: 0)
+      if File.exist?(file.to_s)
+        contracts = Eth::Contract::Initializer.new(file).build_all
+        raise ArgumentError, "No contracts compiled." if contracts.empty?
+        contract = contracts[contract_index].class_object.new
+      else
+        raise ArgumentError, "Cannot find the contract at #{file.to_s}!"
+      end
+      return contract
+    end
+
+    # Creates a contract wrapper from ABI and address.
+    #
     # @param abi [String] contract abi string.
     # @param address [String] contract address.
     # @param name [String] name of contract.
-    # @param contract_index [Number] specify contract.
     # @return [Eth::Contract::Object] Returns the class of the smart contract.
     # @raise [JSON::ParserError] if the json format is wrong.
-    # @raise [ArgumentError] if argument is incorrect.
-    def self.create(file: nil, bin: nil, abi: nil, address: nil, name: nil, contract_index: nil)
-      if File.exist?(file.to_s)
-        contracts = Eth::Contract::Initializer.new(file).build_all
-        raise "No contracts compiled" if contracts.empty?
-        if contract_index
-          contract = contracts[contract_index].class_object.new
-        else
-          contract = contracts.first.class_object.new
+    # @raise [ArgumentError] if ABI, address, or name is missing.
+    def self.from_abi(abi: nil, address: nil, name: nil)
+      unless [address, abi, name].include? nil
+        begin
+          abi = abi.is_a?(Array) ? abi : JSON.parse(abi)
+        rescue JSON::ParserError => e
+          raise e
```

Comment:
```suggestion
        abi = abi.is_a?(Array) ? abi : JSON.parse(abi)
```

if we are rescuing and then only re-raising the same exception, it's equivalent to not having the begin/rescue